--- STM32F40x.py.original	2016-05-22 09:28:04.307258924 -0700
+++ STM32F40x.py	2016-05-22 09:28:41.171345539 -0700
@@ -1189,7 +1189,7 @@
 }
 
 cpu = {
-  # 'name': 'string',
+  'name': 'CM4',
   # 'revision': 'string',
   # 'endian': 'string',
   # 'mpuPresent': boolean,
@@ -1200,7 +1200,7 @@
   # 'itcmPresent': boolean,
   # 'dtcmPresent': boolean,
   # 'vtorPresent': boolean,
-  # 'nvicPrioBits': integer,
+  'nvicPrioBits': 4,
   # 'vendorSystickConfig': boolean,
   # 'deviceNumInterrupts': integer,
 }
